plugins/woocommerce/src/Caches/OrderCountCacheService.php

```diff
@@ -29,6 +32,35 @@ final public function init() {
 		add_action( 'woocommerce_new_order', array( $this, 'update_on_new_order' ), 10, 2 );
 		add_action( 'woocommerce_before_delete_order', array( $this, 'update_on_delete_order' ), 10, 2 );
 		add_action( 'woocommerce_order_status_changed', array( $this, 'update_on_order_status_changed' ), 10, 4 );
+		add_action( 'init', array( $this, 'register_background_caching' ) );
+	}
+
+	/**
+	 * Register background caching for each order type.
+	 *
+	 * @return void
+	 */
+	public function register_background_caching() {
+		$order_types        = wc_get_order_types( 'order-count' );
+		$background_caching = wc_get_container()->get( BackgroundCache::class );
+		foreach ( $order_types as $order_type ) {
+			$background_cache_action = new CacheAction(
+				array(
+					'callback'            => function() use ( $order_type ) {
+						error_log( 'Removing order count cache for ' . $order_type );
+						$this->order_count_cache->remove( $order_type );
+						OrderUtil::get_count_for_type( $order_type );
+					},
+					'force_refresh'       => true,
+					'interval_in_seconds' => HOUR_IN_SECONDS * 12,
```

Comment:
At the time of writing, the expiration for order counts cache is 24 hours.  This forcibly refreshes the cache well before we get to that point so we (hopefully) never see an expired cache and as a failsafe to keep numbers from getting out of sync in case we run into any issues in https://github.com/woocommerce/woocommerce/pull/54034.